daemon/db.c

```diff
@@ -1350,13 +1350,13 @@ void db_set_anchor(struct peer *peer)
 	tal_free(ctx);
 }
 
-bool db_set_visible_state(struct peer *peer)
+void db_set_visible_state(struct peer *peer)
 {
-	const char *errmsg, *ctx = tal_tmpctx(peer);
+	const char *ctx = tal_tmpctx(peer);
 	const char *peerid = pubkey_to_hexstr(ctx, peer->dstate->secpctx, peer->id);
 
 	log_debug(peer->log, "%s(%s)", __func__, peerid);
-	db_start_transaction(peer);
+	assert(peer->dstate->db->in_transaction);
```

Comment:
No problem, it will still fail pretty clearly if you're outside of a transaction so we should be ok :D